server/server.go

```diff
@@ -223,60 +223,59 @@ func (s *Server) githubEvent(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	var pingEvent *github.PingEvent
 	if r.Header.Get("X-GitHub-Event") == "ping" {
-		pingEvent = PingEventFromJSON(ioutil.NopCloser(bytes.NewBuffer(buf)))
+		if pingEvent := PingEventFromJSON(ioutil.NopCloser(bytes.NewBuffer(buf))); pingEvent != nil {
+			mlog.Info("ping event", mlog.Int64("HookID", pingEvent.GetHookID()))
+			return
+		}
 	}
 
 	event := PullRequestEventFromJSON(ioutil.NopCloser(bytes.NewBuffer(buf)))
-	eventIssueComment := IssueCommentFromJSON(ioutil.NopCloser(bytes.NewBuffer(buf)))
-
 	if event != nil && event.PRNumber != 0 {
 		mlog.Info("pr event", mlog.Int("pr", event.PRNumber), mlog.String("action", event.Action))
 		s.handlePullRequestEvent(ctx, event)
 		return
 	}
 
-	pr, err := s.getPRFromComment(ctx, *eventIssueComment)
+	eventData := EventDataFromJSON(ioutil.NopCloser(bytes.NewBuffer(buf)))
+	if eventData == nil || eventData.Action != "created" {
+		s.handleIssueEvent(ctx, event)
+		return
+	}
+
+	pr, err := s.getPRFromEvent(ctx, *eventData)
 	if err != nil {
 		mlog.Error(err.Error())
 		return
 	}
 	var commenter string
-	if eventIssueComment.Comment != nil && eventIssueComment.Comment.User != nil {
-		commenter = eventIssueComment.Comment.User.GetLogin()
+	if eventData.Comment != nil && eventData.Comment.User != nil {
+		commenter = eventData.Comment.User.GetLogin()
 	}
 
-	if eventIssueComment != nil && eventIssueComment.Action == "created" {
-		if strings.Contains(strings.TrimSpace(*eventIssueComment.Comment.Body), "/check-cla") {
-			if err := s.handleCheckCLA(ctx, pr); err != nil {
-				mlog.Error("Error checking CLA", mlog.Err(err))
-			}
+	if eventData.CheckCLA() {
```

Comment:
Maybe we could change all this CHeck explosion into a factory that returns the proper handler to be executed. What do you think? The rationale behind is that as we would increase then umber of commands we would have a ton more of this boilerplate

Reply:
Once we have a better function signature on handlers, we can definitely go to that path. I was thinking the same with you actually. 👍